src/byrow.jl

```diff
@@ -51,31 +51,27 @@ function byrow_helper(df, body)
     (e_body, e_newcols) = byrow_find_newcols(body, Any[])
     e_delimiters = Expr(:(=), :row, :( 1:_N ))
     e_forloop = Expr(:for, e_delimiters, byrow_replace(e_body))
-    return quote
-        _N = length($df[1])
-        _DF = @transform($df, $(e_newcols...))
-        @with _DF $e_forloop
-        _DF
-    end
+    :(_N = length($df[1]); _DF = @transform($df, $(e_newcols...)); @with _DF $e_forloop; _DF)
 end
 
 """
-```julia 
-@byrow!(d, expr) 
-``` 
+```julia
+@byrow!(d, expr)
+```
 
 Act on a DataFrame row-by-row.
 
 Includes support for control flow and `begin end` blocks. Since the
-"environment" induced by `@byrow! df` is implicitly a single row of `df`, 
+"environment" induced by `@byrow! df` is implicitly a single row of `df`,
 use regular operators and comparisons instead of their elementwise counterparts
 as in `@with`. Note that the scope within `@byrow!` is a hard scope.
 
 `byrow!` also supports special syntax for allocating new columns. The syntax
-`@newcol x::Array{Int}` allocates a new column `:x` with an `Array` container
```

Comment:
Well the reason I changed it in the first place is the deprecation of `Array(T, size)`. The macro in the previous version was taking `Container{Eltype}` and turning it into `Container(Eltype, _N)`.  This used to work for both Arrays and DataArrays. Now I could have changed it to make instead `Container{Eltype}(_N)`, which would have been the correct syntax for Arrays but not for  DataArrays. Tldr. the macro was making unwarranted assumptions about constructors, which was why it was able to leave out _N.